net.go

```diff
@@ -90,7 +91,33 @@ func Dial(network, addr string, ctx *Ctx, flags DialFlags) (*Conn, error) {
 func DialTimeout(network, addr string, timeout time.Duration, ctx *Ctx,
 	flags DialFlags) (*Conn, error) {
 	d := net.Dialer{Timeout: timeout}
-	return dialSession(d, network, addr, ctx, flags, nil)
+	if err := prepareCtx(ctx); err != nil {
```

Comment:
The value `ctx` will not be updated.
```suggestion
	if ctx, err := prepareCtx(ctx); err != nil {
```

Reply:
Updated.